core/src/main/scala/org/apache/spark/api/python/PythonWorkerFactory.scala

```diff
@@ -213,7 +213,14 @@ private[spark] class PythonWorkerFactory(pythonExec: String, envVars: Map[String
           daemonPort = in.readInt()
         } catch {
           case _: EOFException =>
-            throw new SparkException(s"No port number in $daemonModule's stdout")
+            val exceptionMessage =
```

Comment:
I think the only thing to do here is to also print the original EOFException.

Reply:
@jiangxb1987 Thanks for your review. But the `DataInputStream` just throws `EOFException` without any detail. The implement of readInt as follows:
```
    public final int readInt() throws IOException {
        int ch1 = in.read();
        int ch2 = in.read();
        int ch3 = in.read();
        int ch4 = in.read();
        if ((ch1 | ch2 | ch3 | ch4) < 0)
            throw new EOFException();
        return ((ch1 << 24) + (ch2 << 16) + (ch3 << 8) + (ch4 << 0));
    }
```